src/hypofuzz/interface.py

```diff
@@ -30,24 +32,39 @@ def has_true_skipif(item: Item) -> bool:
     return False
 
 
+@dataclass
+class CollectionResult:
+    fuzz_targets: list["FuzzProcess"]
+    not_collected: dict[str, dict[str, Any]]
+
+
 class _ItemsCollector:
     """A pytest plugin which grabs all the fuzzable tests at the end of collection."""
 
     def __init__(self) -> None:
         self.fuzz_targets: list[FuzzProcess] = []
+        self.not_collected: dict[str, dict[str, Any]] = {}
+
+    def _skip_because(
+        self, status_reason: str, nodeid: str, kwargs: Optional[dict[str, Any]] = None
+    ) -> None:
+        self.not_collected[nodeid] = {
+            "status_reason": status_reason,
+            **(kwargs or {}),
+        }
 
     def pytest_collection_finish(self, session: pytest.Session) -> None:
         from hypofuzz.hypofuzz import FuzzProcess
 
         for item in session.items:
             if not isinstance(item, pytest.Function):
-                print(f"skipping non-pytest.Function item {item=}")
+                self._skip_because("not_a_function", item.nodeid)
                 continue
             if list(item.iter_markers("skip")):
-                print(f"skipping {item=} due to an @skip mark")
+                self._skip_because("skip", item.nodeid)
                 continue
             if has_true_skipif(item):
-                print(f"skipping {item=} due to a true @skipif mark")
+                self._skip_because("skipif", item.nodeid)
```

Comment:
I think these markers have an optional `reason=` param/attribute; it'd be nice to expose that if it exists.

Reply:
nice call. we're not actually displaying this extra info on the dashboard right now, but we will eventually